component/prometheus.jsonnet

```diff
@@ -0,0 +1,215 @@
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+local params = inv.parameters.rancher_monitoring;
+local namespace = params.namespace;
+
+local instance = params.prometheusInstance;
+local name = 'prometheus-' + instance;
+
+local labels = {
+  'app.kubernetes.io/name': 'prometheus',
+  'app.kubernetes.io/instance': instance,
+  'app.kubernetes.io/managed-by': 'syn',
+};
+
+local matchLabels = {
+  prometheus: instance,
+  role: 'alert-rules',
+};
+
+local nsSelector = {
+  matchLabels: {
+    SYNMonitoring: 'main',
+  },
+};
+
+// The actual prometheus resource for the prometheus operator
+local prometheus = {
+  apiVersion: 'monitoring.coreos.com/v1',
+  kind: 'Prometheus',
+  metadata: {
+    name: name,
+    namespace: namespace,
+    labels: labels,
+  },
+  spec: {
+    externalLabels: {
+      customer: inv.parameters.customer.name,
+      cluster: inv.parameters.cluster.name,
+    },
+    alerting+: {
+      alertmanagers+: [
+        {
+          name: 'alertmanager-' + params.alertmanagerInstance,
+          namespace: namespace,
+          port: 'web',
+        },
+      ],
+    },
+    baseImage: params.images.prometheus.image,
+    tag: params.images.prometheus.tag,
+    nodeSelector+: {
+      'kubernetes.io/os': 'linux',
+    },
+    podMonitorNamespaceSelector+: nsSelector,
+    ruleNamespaceSelector+: nsSelector,
+    serviceMonitorNamespaceSelector+: nsSelector,
+    ruleSelector+: {
+      matchLabels: matchLabels,
+    },
+    replicas: 1,
```

Comment:
It might make more sense to have most of these default values in `default.yaml` to have them in one place, wdyt?

Additionally I'd suggest to set the following additional defaults:
```
evaluationInterval: 5s
scrapeInterval: 10s
```

Reply:
I was thinking about having most of the default values in `default.yml`. The only downside I can see is that somebody could do

```
parameters:
  rancher_monitoring:
    prometheus: ~
```

followed by

```
parameters:
  rancher_monitoring:
    prometheus:
      [their customization]
```

which could possibly lead to non-functional configurations. However, I haven't checked whether the fields we're setting currently are actually required.

tl;dr: I'm fine with moving the defaults to `class/defaults.yml`, if we accept that users may be able to create non-functional Prometheus/Alertmanager objects